TaskManager.java

```diff
@@ -0,0 +1,71 @@
+@Component
+public class TaskManager {
+    @PersistenceContext
+    private EntityManager entityManager;
+    private static TaskManager instance;
+    private LinkedList<Task> queue = new LinkedList<>();
+    private HashSet<Task> running = new HashSet<>();
+    private HashSet<Task> blocked = new HashSet<>();
+    private Thread checker = new Thread(() -> {
+        while (true) {
+            checkRun();
+        }
+    });
+
+    public TaskManager() {
+        instance = this;
+        checker.run();
+    }
+
+    public static TaskManager getInstance() {
+        return instance;
+    }
+
+    @Transactional
+    public void runTask(Task task) {
+        if (Collections.disjoint(task.getDependsOn(), running)) {
+            running.add(task);
```

Comment:
to remove duplicated code extract function addQueneTask(statusQuene HashSet<Task>, t Task)